tests/test_client/tests.py

```diff
@@ -124,7 +123,6 @@ def test_json_serialization(self):
                         client_method = getattr(self.client, method)
                         method_name = method.upper()
                         response = client_method('/json_view/', data, content_type='application/json')
-                        self.assertEqual(response.status_code, 200)
                         self.assertEqual(response.context['data'], expected)
```

Comment:
It may be unclear why `self.assertEqual(response.context['data'], expected)` fails when the `status_code` isn't 200. I would suggest moving this line below the `self.assertContains` assertion. 

The same feedback applies to the other change in this file and also to the last change in `tests/admin_views/tests.py`. Apart from that this looks excellent.

Reply:
Thanks for the review! I've applied the suggested asserting ordering changes.